opensearch-service-domain-cdk/test/stack-composer.test.ts

```diff
@@ -0,0 +1,225 @@
+import {App} from "aws-cdk-lib";
+import {StackComposer} from "../lib/stack-composer";
+import {Template} from "aws-cdk-lib/assertions";
+
+test('Test missing domain name throws error', () => {
+
+    const app = new App({
+        context: {
+            domainName: ""
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+       env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test missing engine version throws error', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: ""
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test invalid engine version format throws error', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "OpenSearch_1.3"
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test ES engine version format is parsed', () => {
```

Comment:
you're only testing w/ one version of engineVersion, 7.10.  You should change the name of the test, or add more versions in here.